src/main/java/seedu/address/model/product/Product.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.model.product;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Product in MyInsuRec.
+ */
+public class Product {
+    public static final String MESSAGE_CONSTRAINTS = "Product names should be alphanumeric";
+    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
```

Comment:
Might be better if product names allow whitespaces also since it's common that insurance products aren't a single connected word